src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -107,12 +107,15 @@ private void OnDiagnosticsUpdated(object? sender, DiagnosticsUpdatedArgs updateA
         public async Task<TReport[]?> HandleRequestAsync(
             TDiagnosticsParams diagnosticsParams, RequestContext context, CancellationToken cancellationToken)
         {
+            context.TraceSource?.TraceInformation($"{this.GetType()} started getting diagnostics");
```

Comment:
see below comments - would be nice if we could just do something like context.Logger.LogInformation(...), or maybe even context.LogInformation() rather than accessing the trace source directly.  that way we could change the implementation later on if we wanted (e.g. if we wanted to also log to the output window when debugging)

Reply:
I agree, I think `context.LogInformation()` is nicer, and means less changes in future if we change something internally.

Reply:
sure.  i'll make that change.